src/features/jserrors/aggregate/index.js

```diff
@@ -197,98 +197,94 @@ export class Aggregate extends AggregateBase {
-      var jsAttributesHash = stringHashCode(stringify(customParams))
-      var aggregateHash = hash + ':' + jsAttributesHash
+      delete params._softNavAttributes // cleanup temp properties from synchronous evaluation; this is harmless when async from soft nav (properties DNE)
+      delete params._softNavFinished
+    } else { // interaction was cancelled -> error should not be associated OR there was no interaction
+      Object.entries(getInfo(this.agentIdentifier).jsAttributes).forEach(([k, v]) => setCustom(k, v))
+      delete params.browserInteractionId
+    }
+    if (localAttrs) Object.entries(localAttrs).forEach(([k, v]) => setCustom(k, v)) // local custom attrs are applied in either case with the highest precedence
 
-      this.aggregator.store(item[0], aggregateHash, params, item[3], customParams)
+    const jsAttributesHash = stringHashCode(stringify(allCustomAttrs))
+    const aggregateHash = bucketHash + ':' + jsAttributesHash
+    this.aggregator.store(type, aggregateHash, params, newMetrics, allCustomAttrs)
 
-      function setCustom (key, val) {
-        customParams[key] = (val && typeof val === 'object' ? stringify(val) : val)
-      }
-    })
-    delete this.errorCache[interaction.id]
+    function setCustom (key, val) {
+      allCustomAttrs[key] = (val && typeof val === 'object' ? stringify(val) : val)
+    }
   }
 
-  onInteractionDiscarded (interaction) {
-    if (!this.errorCache || !this.errorCache[interaction.id] || this.blocked) return
+  onInteractionDone (interaction, wasSaved) {
+    if (!this.bufferedErrorsUnderSpa[interaction.id] || this.blocked) return
 
-    this.errorCache[interaction.id].forEach((item) => {
-      var customParams = {}
-      var globalCustomParams = item[4]
-      var localCustomParams = item[5]
+    this.bufferedErrorsUnderSpa[interaction.id].forEach((item) => {
+      var allCustomAttrs = {}
+      const localCustomAttrs = item[4]
 
-      mapOwn(globalCustomParams, setCustom)
-      mapOwn(interaction.root.attrs.custom, setCustom)
-      mapOwn(localCustomParams, setCustom)
+      mapOwn(interaction.root.attrs.custom, setCustom) // tack on custom attrs from the interaction
+      mapOwn(localCustomAttrs, setCustom)
 
       var params = item[2]
+      if (wasSaved) {
+        params.browserInteractionId = interaction.root.attrs.id
+        if (params._interactionNodeId) params.parentNodeId = params._interactionNodeId.toString()
+      }
       delete params._interactionId
       delete params._interactionNodeId
 
-      var hash = item[1]
-      var jsAttributesHash = stringHashCode(stringify(customParams))
+      var hash = wasSaved ? item[1] + interaction.root.attrs.id : item[1]
+      var jsAttributesHash = stringHashCode(stringify(allCustomAttrs))
       var aggregateHash = hash + ':' + jsAttributesHash
 
-      this.aggregator.store(item[0], aggregateHash, item[2], item[3], customParams)
+      this.aggregator.store(item[0], aggregateHash, params, item[3], allCustomAttrs)
```

Comment:
since you added a method for `storeJsErrorForHarvest`, doesn't this seem duplicative now?

Reply:
storeJsErrorForHarvest (soft nav) has somewhat different flow from on onInteractionDone (old spa), and while they certainly share similarities, I think it'd be too messy to combine them. The intention is to delete onInteractionDone when old spa gets removed. I made a comment for that, work?